pallets/identity/src/lib.rs

```diff
@@ -2003,6 +2007,26 @@ impl<T: Trait> Module<T> {
             })
             .collect::<Vec<_>>()
     }
+
+    /// Checks call permissions and, if successful, returns the caller's account, primary and secondary identities.
+    ///
+    /// TODO: `Context::current_identity_or` fails for origins that do not have an identity. Hence
+    /// secondary keys without an identity are not functional.
+    pub fn ensure_origin_call_permissions(
```

Comment:
Does this still work when accepting authorisations (e.g. JoinIdentity) from a key which doesn't have an identity, where the issuer of the authorisation pays (and should be the identity for CDD purposes)?

Reply:
No, which is why I don't use this function in `join_identity_as_key` and `join_identity_as_identity` dispatchables.

Do you think we should generalise `ensure_call_origin_permissions` to work without an identity?